packages/block/src/header.ts

```diff
@@ -376,6 +385,9 @@ export class BlockHeader {
    * @param height - If this is an uncle header, this is the height of the block that is including it
    */
   async validate(blockchain?: Blockchain, height?: BN): Promise<void> {
+    if (this._common.consensusType() !== 'pow') {
+      throw new Error('block validation is currently only supported on PoW chains')
+    }
```

Comment:
I think this was not a good choice to completely throw on block validation and prevent users to be able to do all the remaining checks. I will change this to just skip the difficulty check.

Reply:
This is OK for this PR, but I think this is what we should do:

We introduce this new `BlockValidator` interface, which does two things: it has `validateConsensus(block: Block)` which allows us to validate consensus of this `Block` (for Ethash, this is to verify the PoW of this block. For Clique, it is to do the POA check). We also add `validateBlock(block: Block)`, which does these checks like validate difficulty, timestamp, etc. For instance, in Clique, it seems that the `difficulty` is used as an extra parameter which doesn't have much to do with difficulty in general. [See the Clique EIP 225](https://eips.ethereum.org/EIPS/eip-225). Note: we probably also want to add ways to input a `BlockHeader` here besides `Block`.

We might want to either figure out what these Consensus methods have in common (probably; timestamp is always strictly larger). But most things are actually Consensus specific: for instance, Clique doesn't have any uncle validation logic, besides that it should be a constant (empty list hash it seems). 

So in general the suggestion is that we remove all PoW consensus/validation logic from `Blockchain` and `Block` and instead encapsulate this in an abstract class `Consensus`, where we can add both Ethash/Clique and possibly more algorithms in the future. So this would be a strategy pattern here, it would make a lot of sense to me.

Looks like these changes are rather important but also breaking, so we should discuss these asap if we want to implement them before V5.

Reply:
I wouldn't want to take this in any more, this is getting too large and too risky for a short-term last minute refactor. This PR wasn't meant to open up this whole "how to implement Clique?" topic but just doing some base preparations. Let's please see how far we'll get with this along the v5 release series, but I rather would want to close this whole PR than going here any further.